sdks/node-ts/src/apache_beam/coders/js_coders.ts

```diff
@@ -0,0 +1,56 @@
+import * as BSON from 'bson';
+import { Writer, Reader } from 'protobufjs';
+import {Coder, CODER_REGISTRY, Context} from './coders'
+import {BoolCoder, DoubleCoder, StrUtf8Coder, VarIntCoder} from './standard_coders'
+
+
+export class BsonObjectCoder<T> implements Coder<T> {
+    static URN = "beam:coder:bsonjs:v1";
+    encode(element: T, writer: Writer, context: Context) {
+        const buff = BSON.serialize(element);
+        writer.bytes(buff);
+    }
+    decode(reader: Reader, context: Context): T {
+        const encoded = reader.bytes();
+        return BSON.deserialize(encoded) as T;
+    }
+}
+CODER_REGISTRY.register(BsonObjectCoder.URN, BsonObjectCoder)
+
+export class GeneralObjectCoder<T> implements Coder<T> {
+    static URN = "beam:coder:genericobjectjs:v1"
+    componentCoders = {
+        'string': new StrUtf8Coder(),
+        'number': new DoubleCoder(),  // TODO(pabloem): What about integers? Do we represent always as doubles?
```

Comment:
Kinda seems like you just want to find a way to use BSON for everything so that it is very native to JS types, and not even bother with standard coders?

Reply:
hmm the issue is that BSON encodes numbers/strings 'wrong' (e.g. `BSON.deserialize(BSON.serialize('abc'))` returns the object `{'0': 'a', '1': 'b', '2': 'c'}`, so that's why I'm encoding those with their own coders. WDYT?